src/engine/strat_engine/blockdev.rs

```diff
@@ -66,10 +67,15 @@ pub fn find_all() -> EngineResult<HashMap<PoolUuid, HashMap<DevUuid, BlockDev>>>
 
         if let Some(bda) = BDA::load(&mut f).ok() {
             let dev = try!(Device::from_str(&devnode.to_string_lossy()));
+            // TODO: Parse MDA and also initialize RangeAllocator with
```

Comment:
`flex_devs` gives the segments that meta dev, thin meta dev, and thin data dev are using across the blockdevs. Putting this info into the RangeAllocator will let us extend these without stomping on their existing allocations.